### PR TITLE
doc: add maxuploadtarget to bitcoin.conf example

### PR DESCRIPTION
**Pull request by jarolrod**
picking up #21499, author has stated they [can't squash](https://github.com/bitcoin/bitcoin/pull/21499#issuecomment-849277632).

This adds the maxuploadtarget option to the `bitcoin.conf` example file. This is useful for those looking to configure their bandwidth utilization.

**Changes from Original PR:**
- squash commits
- fix typo in commit message + reword commit message to be more appropriate
- Implement review suggestions ([1](https://github.com/bitcoin/bitcoin/pull/21499#discussion_r615409982), [2](https://github.com/bitcoin/bitcoin/pull/21499#discussion_r615410337), [3](https://github.com/bitcoin/bitcoin/pull/21499#pullrequestreview-659357756))
- Fix spacing

### CODE REVIEW

File: share/examples/bitcoin.conf [hunk 1]
```diff
@@ -71,6 +71,12 @@
 # configuration option or the addnode RPC, which have a separate limit of 8 connections.
 #maxconnections=
 
+# Maximum upload bandwidth target in MiB per day (e.g. 'maxuploadtarget=1024' is 1 GiB max throughput per day).
```
Comment (theStack) [inline share/examples/bitcoin.conf:74]:
```suggestion
# Maximum upload bandwidth target in MiB per day (e.g. 'maxuploadtarget=1024' is 1 GiB per day).
```

nit: The "max throughput" is kind of redundant to the part before the parantheses, but I'm fine with either version.

**Review by theStack — APPROVED** [review]
ACK 219bbd2f52382c1b4720e8a74ded1a20676c3c9c 📝

File: share/examples/bitcoin.conf (hunk 1 above)
Comment (jonatack) [reply share/examples/bitcoin.conf:74]: I agree, drop "max throughput".

File: share/examples/bitcoin.conf [hunk 2]
```diff
@@ -71,6 +71,12 @@
...
+# This limits the upload bandwidth for those with bandwidth limits. 0 = no limit (default: 0).
+# Limit does not apply to peers with 'download' permission.
```
Comment (jonatack) [inline share/examples/bitcoin.conf:76]: s/Limit/This limit/ or (probably better as it is more explicit): 
```suggestion
# -maxuploadtarget does not apply to peers with 'download' permission.
```

**Comment by jarolrod** [comment]
updated from 219bbd2 -> 947f973 ([pr22080.01](https://github.com/jarolrod/bitcoin/tree/pr22080.01) -> [pr22080.02](https://github.com/jarolrod/bitcoin/tree/pr22080.02), [diff](https://github.com/jarolrod/bitcoin/compare/pr22080.01...jarolrod:pr22080.02?diff=split))

**Changes:**
- Addressed suggestions ([1](https://github.com/bitcoin/bitcoin/pull/22080#discussion_r641540735), [2](https://github.com/bitcoin/bitcoin/pull/22080#discussion_r641582924)) from @theStack and @jonatack (thanks!)

File: share/examples/bitcoin.conf (hunk 2 above)
Comment (jarolrod) [reply share/examples/bitcoin.conf]: Thanks, addressed in  947f973  by using the `-maxuploadtarget` suggestion

File: share/examples/bitcoin.conf (hunk 1 above)
Comment (jarolrod) [reply share/examples/bitcoin.conf]: You're right, addressed in  947f973

**Comment by jonatack** [comment]
ACK 947f9734daab4e47c0abdc6ef7d52812102ecb6b

**Review by theStack — APPROVED** [review]
re-ACK 947f9734daab4e47c0abdc6ef7d52812102ecb6b

**Pull request merged**